app/src/main/java/at/aau/edu/wizards/MainActivity.kt

```diff
@@ -60,15 +70,54 @@ class MainActivity : AppCompatActivity() {
     }
 
     private fun setupUi() {
+        val username = mainViewModel.getUsername()
+        if (username != null) {
+            binding.inputUsername.setText(username)
+            setButtonsEnableStatus(true)
+        } else {
+            setButtonsEnableStatus(false)
+        }
+
+        binding.inputUsername.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
+                // Not required
+            }
+            override fun onTextChanged(typedText: CharSequence, i: Int, i1: Int, i2: Int) {
+                if (typedText.isNotEmpty()) {
+                    setButtonsEnableStatus(true)
+                }
+                else {
+                    setButtonsEnableStatus(false)
+                }
+            }
+
+            override fun afterTextChanged(editable: Editable) {
+                // Not required
+            }
+        })
+
         binding.clientButton.setOnClickListener {
-            showDiscoverFragment()
+            if (binding.clientButton.isEnabled) {
+                val username = binding.inputUsername.text.toString()
```

Comment:
here and below we have a warning, usually that can be fixed giving a different name. Eg `newUserName`.